packages/redis/test/Adaptor.test.js

```diff
@@ -0,0 +1,334 @@
+  });
+  it('should throw field value is not specified', async () => {
+    setMockClient({
+      hGet: async () => {
+        throw new Error('TypeError: Invalid argument type');
+      },
+    });
+    const state = {};
+    try {
+      await hget('animals:1')(state);
+    } catch (error) {
+      expect(error.message).to.eql('TypeError: Invalid argument type');
+    }
+  });
+  it('should expand references', async () => {
+    setMockClient({
+      hGet: async () => 'cat',
+    });
+    const state = { key: 'animals:1', value: 'species' };
+    const result = await hget(
+      s => s.key,
+      s => s.species
+    )(state);
+    expect(result.data).to.eql('cat');
+  });
+});
+
+describe('set', () => {
+  it('should set a value for a key', async () => {
+    setMockClient({
+      set: async () => 'OK',
+    });
+
+    const state = {};
+    const result = await set('name', 'kwaku')(state);
+    expect(result.data).to.eql('OK');
+  });
+  it('should expand references', async () => {
+    setMockClient({
+      set: async () => 'OK',
+    });
+    const state = { key: 'name', value: 'mtuchi' };
+    const result = await set(
+      s => s.key,
+      s => s.value
+    )(state);
+    expect(result.data).to.eql('OK');
+  });
+  it('should throw an error if key and value is not specified', async () => {
+    setMockClient({
+      set: async () => {
+        throw new Error('TypeError: Invalid argument type');
+      },
+    });
+
+    const state = {};
+    try {
+      await set()(state);
+    } catch (error) {
+      expect(error.message).to.eql('TypeError: Invalid argument type');
+    }
+  });
+  it('should throw an error if value is not specified', async () => {
+    setMockClient({
+      set: async () => {
+        throw new Error('TypeError: Invalid argument type');
+      },
+    });
+
+    const state = {};
+    try {
+      await set('patient')(state);
+    } catch (error) {
+      expect(error.message).to.eql('TypeError: Invalid argument type');
+    }
+  });
+  it('should throw an error if key is not specified', async () => {
+    setMockClient({
+      set: async () => {
+        throw new Error('TypeError: Invalid argument type');
+      },
+    });
+
+    const state = {};
+    try {
+      await set('', 'kwaku')(state);
+    } catch (error) {
+      expect(error.message).to.eql('TypeError: Invalid argument type');
+    }
+  });
+});
+
+describe('hset', () => {
+  it('should set fields value for a specified key', async () => {
+    setMockClient({
+      hSet: async () => 1,
+    });
+
+    const state = {};
+    const result = await hset('patient', { name: 'kwaku' })(state);
+    expect(result.data).to.eql(1);
+  });
+  it('should expand references', async () => {
+    setMockClient({
+      hSet: async () => 1,
+    });
+    const state = { key: 'patient', value: { name: 'mtuchi' } };
+    const result = await hset(
+      s => s.key,
+      s => s.value
+    )(state);
+    expect(result.data).to.eql(1);
```

Comment:
>Why 1? 🤔 

After couple of testing i have realized if they matches an existing key the the response will be `1` otherwise it will be `0`

So for example if in the database we already have a key `animal` then we created another hash key `animal:1` then the response will be `1`

Reply:
I like the idea of testing arguments, but i think i need `sinon` package and i am not sure if that's a good idea 🤔 , Do you have other alternative ways of testing arguments ?

Reply:
You may have to verify this for me but I suspect that redis will return 1 if the set was successful, or 0 if the set failed for some reason.

But those responses don't otherwise have value. We don't need to report that back to the user.

It would be more "openfn-y " for us to return nothing if the set was successful, but to throw an exception if the set failed. Like we'll quietly assume the set completed successfully but totally freak out if it does not.

The beauty of these unit tests is that you are showing me the behaviour of the adaptor, and giving me the opportunity to decide whether I think that behaviour is right or wrong. Otherwise I'd have to set up redis and do a bunch of manual testing and who knows if I'd even think to test this?

So this is great process! We're  catching problems in testing before our users encounter them in the field.

Reply:
**For `hset()`** 
- If it's a new key redis will return `1`
- If it's an existing key ti will return `0`

Regarding openfn return nothing, i think it's a good idea to give user feedback when the operation run successfully. It's okay not to mutate state, I can't think of a single usecase for returned response, so i am happy to just return state